pandas/core/frame.py

```diff
@@ -293,10 +293,22 @@
 
       .. versionadded:: 1.2.0
 
-on : label or list
+on : label or list, or Callable
     Column or index level names to join on. These must be found in both
     DataFrames. If `on` is None and not merging on indexes then this defaults
     to the intersection of the columns in both DataFrames.
+    If Callable this will perform a conditional merge. This allows you to
+    provide a custom match condition for the join. The callable takes 2
+    arguments, which represent the left and right sides of the merge, and
+    must return a boolean.
+    ** note: Currently, for callables, the 2 arguments are represented by
```

Comment:
no this is not needed. if we change this we would deprecate or provide a path to change.

Reply:
👍  removed